sympy/assumptions/ask.py

```diff
@@ -154,7 +159,8 @@ def ask(proposition, assumptions=True, context=global_assumptions):
             return True
         if Not(key) in known_facts_dict[local_facts]:
             return False
-    elif local_facts.func is And and all(k in known_facts_dict for k in local_facts.args):
+    elif local_facts.func is And \
+    and all(k in known_facts_dict for k in local_facts.args):
```

Comment:
I prefer to break at the outermost level. Tearing some inner pair of parenthese apart just to avoid the backslash is "letters over structure", which I avoid because it makes understanding the code structure harder.

So I'm going for either

```
    elif (local_facts.func is And and
            all(k in known_facts_dict for k in local_facts.args)):
```

or

```
    elif local_facts.func is And and \
            all(k in known_facts_dict for k in local_facts.args):
```

Here a line break is required because the line is over 80 characters.